torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8352,16 +8353,17 @@ def sample_inputs_ctc_loss(op_info, device, dtype, requires_grad, **kwargs):
     num_char = 20
     target_length = 30
 
-    def make_log_probs(s):
-        t = make_tensor(s, device=device, dtype=dtype)
-        log_probs = t.log_softmax(2).to(device=device, dtype=dtype).detach().requires_grad_(requires_grad=requires_grad)
-        return log_probs
+    # Test ctc_loss gradient behavior for the regime in which it is currently accurate and consistent across devices:
+    # normalized inputs and gradients wrt. unnormalized inputs.
+    # See https://github.com/pytorch/pytorch/issues/52241
+    def make_input(s):
+        return make_tensor(s, device=device, dtype=dtype).detach().requires_grad_(requires_grad=requires_grad)
```

Comment:
This also means that we're running the non gradcheck tests for ctc_loss on unnormalized inputs, is that okay?

Reply:
Good point. These tests do pass, though inputs as documented should be normalized ('log-simplex space'). I think what I can do is just make these sample inputs log_probs, and keep the `gradcheck_wrapper` - since the `log_softmax` op is idempotent.

Reply:
Updated sample inputs to be log_probs. Let's see if this passes CI.

Reply:
Looks like CI mostly passed, at least the jobs that are relevant.